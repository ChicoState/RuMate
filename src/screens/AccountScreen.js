import React from 'react'
import { View, StyleSheet, Text, ScrollView, Image } from 'react-native'
import { Header } from 'react-native-elements'
import firebase from 'firebase'
import Icon from 'react-native-vector-icons/MaterialIcons';
import User from 'react-native-vector-icons/FontAwesome';
import Tile from '../components/Tile'
// import PhotoUpload from 'react-native-photo-upload'

const getDisplayName = () => {
  let username = firebase.auth().currentUser.email.split("@")[0]
  let capital = username[0].toUpperCase()
  username = username.split(username[0])
  return(capital + username[1])
}

const AccountScreen = ({ navigation }) => {
  return(
    <View style={{flex: 1}}>
      <Header
        backgroundColor="#000"
        leftComponent={<Icon name='arrow-back' size={25} color='white' onPress = { () => navigation.navigate('Home')} />}
        centerComponent={{text: "My Account", style: {fontSize: 20, color: 'white'}}}
      />
      <ScrollView style={{height: '100%'}}>
        <Text style={styles.welcomeBanner}>{getDisplayName()}</Text>
        
        <User style = {styles.userLogo}
          name = "user-circle"
          size = {100}
        />
        <Text style={{paddingBottom: '20%'}}>Change photo</Text>
        <Tile style={styles.tile}
          title="Change Display Name"
          color="#111"
          text="Tap to change your display name"
          textColor="white"
          nav={navigation}
          location="ChangeDetails"
        />
        <Tile style={styles.tile}
          title="Change Password"
          color="#111"
          text="Tap to change your password"
          textColor="white"
          nav={navigation}
          location=""
        />
        <Tile style={styles.tile}
          title="Delete Account"
          color="#111"
          text="This permanently deletes your account!"
          textColor="red"
          nav={navigation}
          location=""
        />
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  userLogo: {
    paddingTop: '10%',
    alignSelf: 'center'
  },
  welcomeBanner: {
    fontSize: 30,
    alignSelf: 'center',
    paddingTop: '20%',
  }
});

export default AccountScreen;