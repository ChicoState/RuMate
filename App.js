import React from 'react';
import firebase from 'firebase';
import { StyleSheet } from 'react-native';
import { useScreens } from 'react-native-screens';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from 'react-navigation-stack';
import Icon from 'react-native-vector-icons/FontAwesome';
/* our imports */
import HomeScreen from './src/screens/HomeScreen';
import TasksScreen from './src/screens/TasksScreen';
import AuthScreen from './src/screens/AuthScreen';
import CreateTaskScreen from './src/screens/CreateTaskScreen';
import AddRoommateScreen from './src/screens/AddRoomateScreen';
import CalendarScreen from './src/screens/CalendarScreen';
import BillsScreen from './src/screens/BillsScreen';
import CreateBillScreen from './src/screens/CreateBillScreen';
import CreateRoommateGroupScreen from './src/screens/CreateRoommateGroupScreen';
import MessageScreen from './src/screens/MessagesScreen';
import CreateConversationScreen from './src/screens/CreateConversationScreen';
import ConversationScreen from './src/screens/ConversationScreen';
import InvitationsScreen from './src/screens/InvitationsScreen';
import AccountScreen from './src/screens/AccountScreen';
import PriorityScreen from './src/screens/PriorityScreen';
import ChangeDetailsScreen from './src/screens/ChangeDetailsScreen';

useScreens();
// ################################ //
// Remove to debug errors/warnings  //
   console.disableYellowBox = true; //
// ################################ //
const firebaseConfig = {
  apiKey: "AIzaSyA_ZWMR-MFG_ZHNK4_WuEeHoLP9vzsY_Vk",
  authDomain: "rumate-faaeb.firebaseapp.com",
  databaseURL: "https://rumate-faaeb.firebaseio.com",
  projectId: "rumate-faaeb",
  storageBucket: "",
  messagingSenderId: "484837108351",
  appId: "1:484837108351:web:53a875e9546d00aa2dad4a"
};
// Initialize Firebase
firebase.initializeApp(firebaseConfig);

const tabNavigator = createBottomTabNavigator({
  Home: {
    screen: HomeScreen,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="home" size={20} />
      )
    },
  },
  Tasks: {
    screen: TasksScreen,
    navigationOptions: {
      tabBarLabel: "Tasks",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="tasks" size={20} />
      )
    },
  },
  Bills: {
    screen: BillsScreen,
    navigationOptions: {
      tabBarLabel: "Bills",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="dollar" size={20} />
      )
    },
  },
  Messages: {
    screen: MessageScreen,
    navigationOptions: {
      tabBarLabel: "Messages",
      tabBarIcon: ({ tintColor }) => (
        <Icon name="comments" size={20} />
      )
    },
  }
},
{
  initialRouteName: 'Home',
});

const taskNavigator = createStackNavigator({
  CreateTask: CreateTaskScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const appStackNavigator = createStackNavigator({
  Priority: PriorityScreen,
  Invitations: InvitationsScreen,
  CreateRoommateGroup: CreateRoommateGroupScreen,
  AddRoommate: AddRoommateScreen,
  Home: tabNavigator,
  Tasks: taskNavigator,
  CreateBill: CreateBillScreen,
  TaskCalendar: CalendarScreen,
  Messages: MessageScreen,
  CreateConversation: CreateConversationScreen,
  Conversation: ConversationScreen,
  Account: AccountScreen,
  ChangeDetails: ChangeDetailsScreen

},{
  initialRouteName: 'Home',
  defaultNavigationOptions: {
    header: null
  }
});

const authStack = createStackNavigator({
  Login: AuthScreen,
}, {
  defaultNavigationOptions: {
    header: null
  }
});

const switchNav = createSwitchNavigator({
  App: appStackNavigator,
  Auth: authStack,
}, {
  initialRouteName: 'Auth',
  defaultNavigationOptions: {
    header: null
  }
});

const App = createAppContainer(switchNav);

const styles = StyleSheet.create({
  headerLeft: {
    fontSize: 20
  }
});

export default () => {
  return (
    <App />
  );
}
