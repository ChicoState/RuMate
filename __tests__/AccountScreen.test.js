import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App'
import AccountScreen from '../src/screens/AccountScreen'

describe('<AccountScreen />', () => {
  it('has 1 child', () => {
    <App />
    const tree = renderer.create(<AccountScreen />).toJSON()
    expect(tree.children.length).toBe(1)
  });
  
  it('renders correctly', () => {
    const tree = renderer.create(<AccountScreen />).toJSON()
    expect(tree).toMatchSnapshot()
  });
});