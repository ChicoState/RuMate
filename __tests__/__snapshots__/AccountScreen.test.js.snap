// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
        text="My Account"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        My Account
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginBottom": "1%",
                "marginTop": "20%",
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <Text />
          </View>
        </View>
      </RNGestureHandlerButton>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "paddingBottom": "10%",
            "paddingTop": "2%",
          }
        }
      >
        
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Change Display Name
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to change your display name
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Change Password
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to change your password
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Sign Out
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to log out of RuMate
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            Delete Account
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "red",
                },
              ]
            }
          >
            This permanently deletes your account!
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<InvitationList /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
        text="My Account"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        My Account
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "marginBottom": "1%",
                "marginTop": "20%",
                "transform": Array [
                  Object {
                    "translateY": 0,
                  },
                ],
              }
            }
          >
            <Text />
          </View>
        </View>
      </RNGestureHandlerButton>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 30,
            "paddingBottom": "10%",
            "paddingTop": "2%",
          }
        }
      >
        
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Change Display Name
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to change your display name
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Change Password
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to change your password
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Sign Out
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Tap to log out of RuMate
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#111",
                "borderColor": "#111",
              },
              Object {
                "alignSelf": "center",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "marginHorizontal": 5,
                "marginVertical": 3,
                "paddingHorizontal": 10,
                "width": "95%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "paddingBottom": "5%",
                  "paddingTop": "2%",
                },
                Object {
                  "color": "red",
                },
              ]
            }
          >
            Delete Account
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "red",
                },
              ]
            }
          >
            This permanently deletes your account!
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
